build-system/common/esbuild-babel.js

```diff
@@ -110,9 +148,17 @@ function getEsbuildBabelPlugin(
       preSetup();
 
       build.onLoad({filter: /\.[cm]?js$/, namespace: ''}, async (file) => {
-        const {path} = file;
-        const {contents, hash} = await batchedRead(path);
-        return transformContents(path, contents, hash);
+        const filename = file.path;
+        const contents = await batchedRead(filename);
+        // TODO: make this more efficient.
+        // const babelOptions =
+        //   babel.loadOptions({
```

Comment:
We just need 1 top-level call per `callerName`.

Reply:
Ah. So I'll get the options once per `build`, and then override the filename / sourceFilename on each read.

Reply:
Yah, that should work. The reason this is ridiciulously slow is that babel supports "closest to file" options resolution, meaning you could have a `sub/babel.config.js` file and files inside `sub` will  use that instead of the root `/babel.config.js`. But, we only have a root config, so no need for per-file lookup.